python/pyspark/ml/torch/tests/test_distributor.py

```diff
@@ -0,0 +1,193 @@
+
+        self.sc = SparkContext("local-cluster[2,2,1024]", class_name, conf=conf)
+        self.spark = SparkSession(self.sc)
+
+    def tearDown(self) -> None:
+        os.unlink(self.tempFile.name)
+        self.spark.stop()
+
+    def test_get_num_tasks_locally(self) -> None:
+        succeeds = [1, 2]
+        fails = [4, 8]
+        for num_processes in succeeds:
+            with self.subTest():
+                expected_output = num_processes
+                distributor = TorchDistributor("pytorch", num_processes, True, True)
+                self.assertEqual(distributor._get_num_tasks(), expected_output)
+
+        for num_processes in fails:
+            with self.subTest():
+                with self.assertWarns(RuntimeWarning):
+                    distributor = TorchDistributor("pytorch", num_processes, True, True)
+                    distributor.num_processes = 3
+
+
+class TorchDistributorDistributedUnitTests(unittest.TestCase):
+    def setUp(self) -> None:
+        class_name = self.__class__.__name__
+        self.tempFile = tempfile.NamedTemporaryFile(delete=False)
+        self.tempFile.write(
+            b'echo {\\"name\\": \\"gpu\\", \\"addresses\\": [\\"0\\",\\"1\\",\\"2\\"]}'
+        )
+        self.tempFile.close()
+        # create temporary directory for Worker resources coordination
+        self.tempdir = tempfile.NamedTemporaryFile(delete=False)
+        os.unlink(self.tempdir.name)
+        os.chmod(
+            self.tempFile.name,
+            stat.S_IRWXU | stat.S_IXGRP | stat.S_IRGRP | stat.S_IROTH | stat.S_IXOTH,
+        )
+        conf = SparkConf().set("spark.test.home", SPARK_HOME)
+
+        conf = conf.set("spark.worker.resource.gpu.discoveryScript", self.tempFile.name)
+        conf = conf.set("spark.worker.resource.gpu.amount", "3")
+        conf = conf.set("spark.task.cpus", "2")
+        conf = conf.set("spark.task.resource.gpu.amount", "1")
+        conf = conf.set("spark.executor.resource.gpu.amount", "1")
+
+        self.sc = SparkContext("local-cluster[2,2,1024]", class_name, conf=conf)
+        self.spark = SparkSession(self.sc)
+
+    def tearDown(self) -> None:
+        os.unlink(self.tempFile.name)
+        self.spark.stop()
+
+    def test_get_num_tasks_distributed(self) -> None:
+        inputs = [(1, 8, 8), (2, 8, 4), (3, 8, 3)]
+
+        for spark_conf_value, num_processes, expected_output in inputs:
+            with self.subTest():
+                self.spark.sparkContext._conf.set(
+                    "spark.task.resource.gpu.amount", str(spark_conf_value)
+                )
+                distributor = TorchDistributor("pytorch", num_processes, False, True)
+                self.assertEqual(distributor._get_num_tasks(), expected_output)
+
+        self.spark.sparkContext._conf.set("spark.task.resource.gpu.amount", "1")
+
+
+if __name__ == "__main__":
+    from pyspark.ml.torch.tests.test_distributor import *  # noqa: F401,F403 type: ignore
```

Comment:
Why we need to add F403? I saw other test files only include F401.

Reply:
`mypy` raises errors otherwise